internal/ledgerutil/compare.go

```diff
@@ -39,8 +44,22 @@ func Compare(snapshotDir1 string, snapshotDir2 string, outputFile string) (count
 		return 0, errors.New("both snapshots public state hashes are same. Aborting compare")
 	}
 
+	// Output directory creation
+	outputDirName := filepath.Base(snapshotDir1) + "_and_" + filepath.Base(snapshotDir2) + "_compared"
+	outputDirPath := filepath.Join(outputDirLoc, outputDirName)
+
+	// Check for existing output directory
+	if _, err := os.Stat(outputDirPath); !os.IsNotExist(err) {
+		return 0, errors.Errorf("%s already exists in %s. Choose a different location or remove the existing results. Aborting compare", outputDirName, outputDirLoc)
+	}
+
+	err = os.Mkdir(outputDirPath, 0o755)
```

Comment:
- This is not a correct check. Any general I/O error will cause to proceed inside this `if` block - which is not your intention. You can use use[ this function](https://github.com/hyperledger/fabric/blob/240cf0e182938a09832ba4162c3c229c81848e3f/internal/fileutil/fileutil.go#L120) here.

- Even simpler would be to simply use [this function](https://github.com/hyperledger/fabric/blob/240cf0e182938a09832ba4162c3c229c81848e3f/internal/fileutil/fileutil.go#L77) which will create the whole path and will return to you whether the dir is empty and you can simply check that the return value is `true`

Reply:
Thanks for the tip